lib/factorial.rb

```diff
@@ -1,4 +1,25 @@
 # Computes factorial of the input number and returns it
 def factorial(number)
-  raise NotImplementedError
+
+  raise ArgumentError if number == nil
+
+  count = (number -1)
+  sum = count * number
+  if number == 1 || number == 0
+    sum = 1
+    return sum
+  end
+
+
+  while count > 0
```

Comment:
You could get rid of the conditional statements in this loop by simply making the loop condition be `while count > 1`